tests/semantic_analyser.cpp

```diff
@@ -5603,6 +5603,15 @@ kprobe:f { fail("always fail"); }
            ~~~~~~~~~~~~~~~~~~~
 )" });
   test(R"(kprobe:f { if (false) { fail("always false"); } })");
+  test(
```

Comment:
Ah, these are some of the missing tests I was complaining were missing in the typeinfo PR. Looks like they just ended up here.